### ISSUE TITLE
XS version of one() returns true when passed an empty list

### ISSUE THREAD

**Issue by randyl** [houseabsolute/List-SomeUtils#3]
The docs state:
> For an empty LIST, "one" returns false

However, I'm using perl 5.18, List::SomeUtils 0.56 and macOS 10.12, and I get the following:
```
$ perl -MList::SomeUtils -le 'print "defined" if List::SomeUtils::one { defined($_) } ();'
defined
$
```
The PP version works as expected:
```
$ perl -MList::SomeUtils::PP -le 'print "defined" if List::SomeUtils::PP::one { defined($_) } ();'
$
```

**Comment by randyl** [comment]
This issue also affect List::MoreUtils. I've filed a ticket for it here: https://github.com/perl5-utils/List-MoreUtils/issues/31

**Comment by autarch** [comment]
Definitely a bug. I started adding tests for empty list behavior and found a bunch of other bugs, so far in the PP code, but probably also in some XS! I suspect these all apply to LMU as well.

**Comment by petdance** [comment]
Are you submitting 'em over to LMU?

**Comment by petdance** [comment]
Are you submitting over to LMU?  I'd be very interested to see what they are.  I'm surprised it's taken this long for anyone to discover this one.

**Comment by autarch** [comment]
No, I'm not going to submit to LMU, but there will be a branch that I push shortly with new tests and changes to get them to pass (followed by a branch in the XS repo).

**Comment by autarch** [comment]
This is fixed in the latest release, v0.56. The other bugs I noted were in the pure Perl implementation. I did a trial release of LSU with those fixes, since there's a bit of backwards incompatibility there. The XS code was doing the right thing in those cases.
